templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java

```diff
@@ -88,64 +102,42 @@
         return configs;
     }
 
-    private JSONArray getVMsFromProjectAPI(final AzureTREAuthenticatedUser user) {
-
+    private JSONArray getVMsFromProjectAPI(final AzureTREAuthenticatedUser user) throws GuacamoleException {
         final JSONArray virtualMachines;
-
-        // Todo: Implement / Uncomment when the relevant API call is available for consumption
-        // https://github.com/microsoft/AzureTRE/issues/558
-    /*
-    try {
-      SSLContextBuilder builder = new SSLContextBuilder();
-      builder.loadTrustMaterial(null, new TrustSelfSignedStrategy());
-      SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(builder.build());
-      CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslsf).build();
-
-      try {
-
-        URI projectUri = new URI(System.getenv("PROJECT_URL"));
-
-        // specify the host, protocol, and port
-        URIBuilder uriBuilder = new URIBuilder();
-
-        uriBuilder.setScheme(projectUri.getScheme()).setHost(projectUri.getHost()).setPort(projectUri.getPort())
-            .setPath("/api/userserviceinstances").setParameter("ResourceGroupId", System.getenv("RESOURCE_GROUP_ID"))
-            .setParameter("ResourceType", "virtual-desktop-guacamole");
-
-        URI uri = uriBuilder.build();
-        HttpGet httpget = new HttpGet(uri);
-        httpget.addHeader("Authorization", "Bearer " + user.getAccessToken());
-
-        CloseableHttpResponse httpResponse = httpclient.execute(httpget);
-
-        String json = EntityUtils.toString(httpResponse.getEntity());
-        if (json.length() != 0) {
-          virtualMachines = new JSONArray(json);
-        } else {
-          virtualMachines = new JSONArray();
+        try {
```

Comment:
Overall, seems a bit complicated than the "native" java 11 method of using HttpClient. Something like this (not 100% complete):
```java
        String url = String.format("%s/api/workspace-services/%s/user-resources", System.getenv("API_URL"), System.getenv("SERVICE_ID"));

        // create a client
        var client = HttpClient.newHttpClient();

        // create a request
        var request = HttpRequest.newBuilder(URI.create(url))
                .header("accept", "application/json")
                .header("Authorization", "Bearer " + user.getAccessToken())
                .timeout(Duration.ofSeconds(5))
                .build();

        // use the client to send the request
        var response = client.send(request, HttpResponse.BodyHandlers.ofString());

        // the response:
        if (!response.body().isBlank()) {
            final JSONObject result = new JSONObject(response.body());
            virtualMachines = result.getJSONArray("userResources");
        } else {
            virtualMachines = new JSONArray();
        }
```

Reply:
Fixed. Actually worked pretty much out of the box.

Reply:
I fixed all the other comments and tested e2e (created new bundles and guacamole image, created new ws, new guacamole, new vm, saw it connecting)